cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -220,86 +218,79 @@ func (cm *certManagerClient) createObj(o unstructured.Unstructured) error {
-func (cm *certManagerClient) hasWebhook() (bool, error) {
-	// Checks if the cert-manager web-hook already exists, if yes, no additional images are required
-	webhook, err := cm.getWebhook()
-	if err != nil {
-		return false, errors.Wrap(err, "failed to check if the cert-manager web-hook exists")
-	}
-	if webhook != nil {
-		return true, nil
-	}
-	return false, nil
-}
+// waitForAPIReady will attempt to create the cert-manager 'test assets' (i.e. a basic
+// Issuer and Certificate).
+// This ensures that the Kubernetes apiserver is ready to serve resources within the
+// cert-manager API group.
+func (cm *certManagerClient) waitForAPIReady(ctx context.Context, log logr.Logger) error {
+	// Waits for for the cert-manager web-hook to be available.
+	log.Info("Waiting for cert-manager to be available...")
 
-// isWebhookAvailable returns true if the cert-manager web-hook has the condition type:Available with status:True.
-// This is required to check the web-hook is working and ready to accept requests.
-func (cm *certManagerClient) isWebhookAvailable(webhook *unstructured.Unstructured) (bool, error) {
-	conditions, found, err := unstructured.NestedSlice(webhook.Object, "status", "conditions")
+	testObjs, err := getTestResourcesManifestObjs()
 	if err != nil {
-		return false, errors.Wrap(err, "invalid cert-manager web-hook: failed to get conditions")
-	}
-
-	// if status.conditions does not exists, we assume the web-hook is still starting
-	if !found {
-		return false, nil
+		return err
 	}
 
-	// look for the condition with type:Available and status:True or return false
-	for _, condition := range conditions {
-		conditionMap, ok := condition.(map[string]interface{})
-		if !ok {
-			return false, errors.Wrap(err, "invalid cert-manager web-hook: failed to parse conditions")
-		}
-
-		conditionType, ok := conditionMap["type"]
-		if !ok {
-			return false, errors.Wrap(err, "invalid cert-manager web-hook: there are conditions without the type field")
-		}
+	for i := range testObjs {
+		o := testObjs[i]
+		log.V(5).Info("Creating", logf.UnstructuredToValues(o)...)
 
-		if conditionType != "Available" {
-			continue
+		// Create the Kubernetes object.
+		// This is wrapped with a retry as the cert-manager API may not be available
+		// yet, so we need to keep retrying until it is.
+		if err := cm.pollImmediateWaiter(waitCertManagerInterval, cm.getWaitTimeout(), func() (bool, error) {
+			if err := cm.createObj(o); err != nil {
+				log.V(5).Info("Failed to create cert-manager test resource", logf.UnstructuredToValues(o)...)
+				return false, nil
+			}
+			return true, nil
+		}); err != nil {
+			return err
 		}
+	}
+	for i := range testObjs {
+		o := testObjs[i]
+		log.V(5).Info("Deleting", logf.UnstructuredToValues(o)...)
 
-		conditionStatus, ok := conditionMap["status"]
-		if !ok {
-			return false, errors.Wrapf(err, "invalid cert-manager web-hook: there %q condition does not have the status field", "Available")
+		cl, err := cm.proxy.NewClient()
```

Comment:
nit.
what about creating cm.deleteObj (for simmetry with cm.createObj)

Reply:
Can do - but it seems `createObj` exists in order to handle adding the label, whereas Delete does not have this requirement. Happy to make the change, but it'll basically be a one-liner function :)

Reply:
we can move inside the function the creation of the client...

Reply:
Do we want to create a new client for each call to `createObj` and `deleteObj`? Or pass it in as an argument?

Reply:
we can move inside the function the creation of the client...
the client instance is cached. so fine to create 1 per call and keep the method signature clean

Reply:
done